### ISSUE TITLE
Can't set NIC InterfaceType in ec2 (need boto3?)

### ISSUE THREAD

**Issue by loveshack** [gc3-uzh-ch/elasticluster#638]
I'd like to set the NIC InterfaceType for ec2 instances.  It looks as if that's not possible with boto2, and it would need a port to boto3 (or is there a way round it)? I think at least tenancy is in the same boat.
It's not clear to me what would be involved in using boto3, or whether it's sensible to try to add features to boto methods.

**Comment by riccardomurri** [comment]
Is this related to https://github.com/gc3-uzh-ch/elasticluster/issues/506 ?

**Comment by riccardomurri** [comment]
Regarding the port to Boto3:

- it's definitely something that needs to be done, sooner or later -- boto2 receives no upgrades.
- but the API has changed almost completely so ElastiCluster's "provider" module needs to be rewritten almost from scratch.

I won't start working on this until 1.3.0 is out, which should happen soon-ish now (I just want to merge a couple of pending PRs before release) but of course if somebody with better understanding of Boto and the EC2 API would like to contribute code, I won't oppose that :-)

**Comment by loveshack** [comment]
> Is this related to https://github.com/gc3-uzh-ch/elasticluster/issues/506 ?

No, as I commented on that one.  It's for the EFA interface (as opposed
to ENA).  That's the new "low latency" one (still an order of magnitude
off what you expect from HPC fabrics)
https://aws.amazon.com/blogs/aws/now-available-elastic-fabric-adapter-efa-for-tightly-coupled-hpc-workloads/.

**Comment by loveshack** [comment]
The doc suggests you can mix boto3 and boto and replace bits piecemeal,
but it wasn't clear to me how that would work from a very quick look.

**Comment by loveshack** [comment]
I'll probably have a chance to look into this, in case anyone else is considering it. If so, I'm happy to collaborate or leave it to someone more expert. Ping me as appropriate.

**Comment by riccardomurri** [comment]
I'm not working on this now, and I don't think anyone else is.  If you have time to take a shot, I'll be very glad to get a PR!  Thank you very much!

**Comment by loveshack** [comment]
Looking at this, I wonder if there's a good reason to use boto3 directly, rather than going through libcloud. Info/opinions welcome from anyone interested who's watching.
